src/datasets/arrow_dataset.py

```diff
@@ -226,171 +238,205 @@ def _get_output_signature(dataset: "Dataset", collate_fn: Callable, collate_fn_a
                 background while the model is training.
+            error_on_missing (:obj:`bool`, default ``True``): Whether to return an error if one of the requested columns
+                is missing. Should be set to `False` if columns or label_cols are autodetected from a model.
 
         Returns:
             :class:`tf.data.Dataset`
         """
-
+        # region Pre-checks and warnings
         if config.TF_AVAILABLE:
             import tensorflow as tf
         else:
             raise ImportError("Called a Tensorflow-specific function but Tensorflow is not installed.")
 
+        if isinstance(tf.distribute.get_strategy(), tf.distribute.TPUStrategy):
+            logger.warning(
+                "Note that to_tf_dataset() loads the data with a generator rather than a full tf.data "
+                "pipeline and is not compatible with remote TPU connections. If you encounter errors, please "
+                "try using a TPU VM or, if your data can fit in memory, loading it into memory as a dict of "
+                "Tensors instead of streaming with to_tf_dataset()."
+            )
+        # endregion
+
+        # region Argument and default value handling
+        if collate_fn is None:
+            # Set a very simple default collator that just stacks things together
+            collate_fn = minimal_tf_collate_fn
         if collate_fn_args is None:
             collate_fn_args = {}
-
         if label_cols is None:
             label_cols = []
         elif isinstance(label_cols, str):
             label_cols = [label_cols]
         elif len(set(label_cols)) < len(label_cols):
             raise ValueError("List of label_cols contains duplicates.")
-        if not columns:
-            raise ValueError("Need to specify at least one column.")
-        elif isinstance(columns, str):
-            columns = [columns]
-        elif len(set(columns)) < len(columns):
-            raise ValueError("List of columns contains duplicates.")
-        if label_cols is not None:
-            cols_to_retain = columns + label_cols
+        if columns:
+            if isinstance(columns, str):
+                columns = [columns]
+            elif len(set(columns)) < len(columns):
+                raise ValueError("List of columns contains duplicates.")
+            if not label_cols:
+                post_collate_cols_to_retain = columns
+            else:
+                post_collate_cols_to_retain = list(set(columns + label_cols))
         else:
-            cols_to_retain = columns
-        if "label" in cols_to_retain or "labels" in cols_to_retain or "label_ids" in cols_to_retain:
-            cols_to_retain += ["labels", "label", "label_ids"]  # Don't accidentally drop any labels with other names!
-        cols_to_retain = list(set(cols_to_retain))  # Remove any duplicates
-
+            post_collate_cols_to_retain = None
```

Comment:
_get_output_signature expects a list of columns, not None

Reply:
Sorry, forgot to add the logic there. Fixing!